bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java

```diff
@@ -153,14 +156,42 @@ public void unsetPersistentStateRestorer(PersistentStateRestorer persistentState
 	public String getName() {
 		return "mysql";
 	}
+	
+	/**
+	 * 
+	 * @param i
+	 * @return
+	 */
+	private String getItemType(Item i) {
+		Item item = i;		
+		if(i instanceof GroupItem){
+			item = ((GroupItem) i).getBaseItem();
+			if(item == null){//if GroupItem:<ItemType> is not defined in *.items using StringType
+				logger.debug("mySQL: Did not find ItemType for {}, 'GroupItem:<ItemType>' is not defined in *.items File. Try to use first Child-Members ItemType now.", i.getName());
+				item = ((GroupItem) i).getMembers().get(0);
+				if(item == null){
+					logger.debug("mySQL: ItemType NOT found! Use STRINGITEM for {}", i.getName());
```

Comment:
A bit confused, does this make sense? When item is null, from where should I get a ItemType? I haven't one at this time, isn't it?
Debug logs, can be more clear, did you mean this? 
my new suggestion:
```
	private String getItemType(Item i) {
		Item item = i;		
		if(i instanceof GroupItem){
			item = ((GroupItem) i).getBaseItem();
			if(item == null){//if GroupItem:<ItemType> is not defined in *.items using StringType
				logger.debug("mySQL: No GroupItem ItemType found for {}, 'GroupItem:<ItemType>' is not defined in *.items File.", i.getName());
				item = ((GroupItem) i).getMembers().get(0);
				if(item == null){
					logger.debug("mySQL: No ItemType found for first Child-Member of GroupItem {}, use ItemType STRINGITEM ({}) as Fallback", i.getName(), sqlTypes.get("STRINGITEM"));
					return sqlTypes.get("STRINGITEM");
				}
			}
		}
		String itemType = item.getClass().getSimpleName().toUpperCase();
		if(sqlTypes.get(itemType) == null){
			logger.debug("mySQL: No sqlType found for ItemType {}, use ItemType STRINGITEM ({}) as Fallback for {}", itemType, sqlTypes.get("STRINGITEM"), i.getName());
			return sqlTypes.get("STRINGITEM");
		}

		logger.debug("mySQL: Use ItemType {} ({}) for Item {}", itemType, sqlTypes.get(itemType), itemType, i.getName());
		return sqlTypes.get(itemType);
	}
```